parallel-install/pkg/installation/installation_test.go

```diff
@@ -0,0 +1,352 @@
+package installation
+
+import (
+	"context"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/components"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/config"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/engine"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes/fake"
+
+	//"k8s.io/client-go/kubernetes/fake"
+	"log"
+	"testing"
+	"time"
+)
+
+const cancelTimeout = 150 * time.Millisecond
+const quitTimeout = 250 * time.Millisecond
+
+func TestInstallation_StartKymaInstallation(t *testing.T) {
+	t.Parallel()
+
+	i := newInstallation()
+
+	t.Run("should install Kyma", func(t *testing.T) {
+		kubeClient := fake.NewSimpleClientset()
+
+		hc := &mockHelmClient{}
+		provider := &mockProvider{
+			hc: hc,
+		}
+		overridesProvider := &mockOverridesProvider{}
+		eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+			WorkersCount: 2,
+			Log:          log.Printf,
+		})
+
+		err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("should fail to install Kyma prerequisites", func(t *testing.T) {
+		t.Run("due to cancel timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 200,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Kyma prerequisites installation failed due to the timeout")
```

Comment:
Please also add time-based assertions with a comment as in "due to quit timeout" test case